test/TestUtil.js

```diff
@@ -30,6 +29,30 @@ exports.promiseLater = function (time, func) {
     }));
 };
 
+/**
+ * Returns rejection reason if rejected, otherwise returns a dummy error.
+ */
+exports.getRejectionReasonOrDummy = async function (object, asyncMethodName, ...params) {
+    try {
+        await object[asyncMethodName](...params);
+        return new Error('dummy error');
```

Comment:
we should throw here to make the test fail in case the method above does not throw
```suggestion
        throw new Error('Expected the call the throw, but it didn't.');
```

Reply:
Similar comments apply to `getThrownErrorOrDummy` below

Reply:
it is the same thing since the error is catched and returned